src/app/src/components/EditableTitle.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import {
+    Flex,
+    IconButton,
+    Editable,
+    EditableInput,
+    EditablePreview,
+    ButtonGroup,
+} from '@chakra-ui/react';
+import { IoMdCreate, IoIosCheckmark, IoIosClose } from 'react-icons/io';
+
+import { renameVisualization } from '../redux/visualizations.actions';
+
+function EditableControls({ isEditing, onSubmit, onCancel, onEdit }) {
+    return isEditing ? (
+        <ButtonGroup
+            alignItems='center'
+            ml={4}
+            onClick={e => e.stopPropagation()}
+        >
+            <IconButton
+                size='sm'
+                isRound
+                icon={<IoIosCheckmark />}
+                onClick={onSubmit}
+                fontSize='2xl'
+            />
+            <IconButton
+                size='sm'
+                isRound
+                icon={<IoIosClose />}
+                onClick={onCancel}
+                fontSize='2xl'
+            />
+        </ButtonGroup>
+    ) : (
+        <Flex
+            alignItems='center'
+            ml={4}
+            minWidth='32px'
+            onClick={e => e.stopPropagation()}
+        >
+            <IconButton
+                size='sm'
+                isRound
+                icon={<IoMdCreate />}
+                onClick={onEdit}
+                display='none'
+                _groupHover={{ display: 'flex' }}
+            />
+        </Flex>
+    );
+}
+
+function EditableTitle({ title, index }) {
+    const dispatch = useDispatch();
+
+    const onSubmit = value => {
+        dispatch(renameVisualization({ title: value, index }));
+    };
+
+    // 'stopPropagation' needs to be used on several subcomponents to prevent
+    // clicks on the buttons and within the input from navigating the user
+    // to the saved visualization.
+    return (
+        <Editable
+            textAlign='left'
+            defaultValue={title}
+            color='rgb(39, 55, 63)'
+            fontSize='2xl'
+            isPreviewFocusable={false}
+            submitOnBlur={false}
```

Comment:
The one issue with this is that it then also submits on cancel (since the input gets blurred). There doesn't seem to be a good way to work around this while having a cancel button. I could simply remove the cancel button, and make any title changes made automatically permanent.

Reply:
Hmm, that makes sense. I assumed it was explicit since the prop was already added. I would say we should treat this similar to how Google Docs allows document titles to be added. Let's drop the cancel and keep the submit on blur.